### ISSUE TITLE
1.13 update

### ISSUE THREAD

**Issue by galacticwarrior9** [TheBusyBiscuit/Slimefun4#769]
Is a 1.13 update planned in the foreseeable future? Slimefun is one of my server's main plugins, and a lack of an update to 1.13 is preventing us from updating server versions altogether. 

Best regards,
galacticwarrior

**Comment by runelynx** [comment]
We decided to upgrade without it just so players can start enjoying 1.13. We'll have to add in some shrubs and trees later, and reconnect the API. It always eventually gets the update, it's just one of the slower plugins. probably extremely painful to update in 1.13 :(

**Comment by TheBusyBiscuit** [comment]
I cannot say how difficult it us to update to 1.13 as I havent been ab active part of this project since years...
But it all comes down to one thing:
Everybody wants an update but it does not seem like anyone is interested in updating the Project.

**Comment by Poslovitch** [comment]
Updating this plugin to 1.13 **is** difficult. As Slimefun heavily relies on blocks, materials and all that kind of stuff, it's really hard to tell how much has to be redone - because we're talking about redoing, not just updating pieces of code.

I will, personally, no longer contribute on Slimefun's codebase. And if I ever do it, this not going to be "Slimefun4" anymore, rather a weird kind of fork where I'd have wiped out everything. Slimefun4 has bugs and is hard to maintain due to its design and to the fact that it is supporting more than a single version at once. If someone were to update Slimefun4, I'd recommend him to not care about backward compatibility. *Keep the plugin compatible with the latest version of Minecraft __only__.*

So, in my opinion, don't expect Slimefun4 to be updated any time soon. And, more likely, don't expect it to be updated at all.

**Comment by SoSeDiK** [comment]
I fully agree with @Poslovitch. Updating Slimefun (& CS-CoreLib) isn't that hard, but it's really time consuming. It can and probably will take more than the one or two whole days of working (on condition that we'll drop support for all older versions of Minecraft and, to be honest, we must do it). In addition, many tests should be done before pushing changes to public. Also, I should note that Slimefun by itself has many bugs. It'll be better to rewrite it from the beginning trying to prevent those bugs, with item hashes, MySQL support and other features that 1.13.1 (*current up to date version) allows. Personally, I'd like to add an optional resource pack feature with Minecraft's models and packets (that includes items and blocks), maybe with sounds. Actually, that is what I already started to do.

**Comment by PrzemoVi** [comment]
https://github.com/TheBusyBiscuit/Slimefun4/pull/756

**Comment by SoSeDiK** [comment]
@PrzemoVi, didn't test it, but I can say that my changes are pretty different in some parts. Is everything (I mean EVERYTHING :D) working? Also, does it support migration from 1.12.2 to 1.13.1 without clearing the plugin's data? If no, you probably should make an item recognition system. I think simple way with checking lores would be enough. Like for example GOLDEN_MILK with a name: "&6Golden Milk"; and lore: "&7Just a great drink", "&0sf:golden_milk". And later check that item with the last lore. Just loop through the all lores, and check if it starts with a "&0sf:" part. This will allow renaming, enchanting Slimefun items and will make all of the items compatible with others plugins like CrazyEnchantments.

**Comment by SoSeDiK** [comment]
This should be discussed with a community although.

**Comment by PrzemoVi** [comment]
Migration from 1.12 to 1.13 should work.
Everything is working, except enchanted furnaces (I'm working on it)

**Comment by TheBusyBiscuit** [comment]
Had some free time (actually just nothing to do at work right now...)
So I went ahead and dug through 1.13.1 NMS Code to find the updated method/field names and map them correspondingly in the NMS-dependent classes.
https://github.com/TheBusyBiscuit/CS-CoreLib/commit/b9e81d7e6625903fd52f82fc8ea8592a74439c74

Updated Enums will follow soon and then there will be a dev build available shortly:
https://thebusybiscuit.github.io/builds/TheBusyBiscuit/CS-CoreLib/master/

**Comment by TheBusyBiscuit** [comment]
CS-CoreLib _should_ work on 1.13.1 now too, as well as 1.9-1.12

I know it is hard to test without a Slimefun Version for MC 1.13 but it is available for people to try and test now nonetheless.
And tbh, CS-CoreLib is the hard part as it handles NMS Code, Slimefun is easy to follow up after this.
https://thebusybiscuit.github.io/builds/TheBusyBiscuit/CS-CoreLib/master/#22

**Comment by TheBusyBiscuit** [comment]
Experimental builds for MC 1.13.1 for Slimefun4 and CS-CoreLib are now available:

https://thebusybiscuit.github.io/builds/

**Comment by ZevBellringer** [comment]
Im playing on 1.13 on mc.barbercraft.net, The server has been updates to 1.13, and is using SF4.   I cannot craft certain items on the Automated Crafting Table - electromagnets, motors, heating coils and various other stuff to do with electrics. You put the correct items in in the correct pattern, enebale the table, but  nothing happens, You can hand fabricate these items, using the same ingrediants,  on the manual Enchanched Crafting Chamber, but  not on the electric one. The server owner assures me we are using buld #31 CScoreliband #64  master SF 4,. Where do we go from here ?

**Comment by PrzemoVi** [comment]
Every pattern that uses "head" items will not work in Automated Crafting Table.
We`ll try to fix this.
